processing/src/main/java/io/druid/query/groupby/epinephelinae/ConcurrentGrouper.java

```diff
@@ -238,13 +259,25 @@ public void reset()
       throw new ISE("Grouper is closed");
     }
 
-    return Groupers.mergeIterators(
-        sorted && isParallelSortAvailable() ? parallelSortAndGetGroupersIterator() : getGroupersIterator(sorted),
-        sorted ? keyObjComparator : null
-    );
+    final List<Iterator<Entry<KeyType>>> sortedIterators = sorted && isParallelizable() ?
+                                                           parallelSortAndGetGroupersIterator() :
+                                                           getGroupersIterator(sorted);
+
+    // Parallel combine is used only when data are spilled. This is because ConcurrentGrouper uses two different modes
+    // depending on data are spilled or not. If data is not spilled, all inputs are completely aggregated and no more
+    // aggregation is required.
+    if (sorted && spilling && isParallelizable()) {
+      // First try to merge dictionaries generated by all underlying groupers. If it is merged successfully, the same
+      // merged dictionary is used for all combining threads
+      final List<String> mergedDictionary = mergeDictionary();
```

Comment:
merge/merged -- choose one

Reply:
Maybe something like "tryCreateMergedDictionary" or "tryMergeDictionary"

Reply:
Changed to tryMergeDictionary